queue/priority_queue.go

```diff
@@ -94,6 +107,7 @@ func (items *priorityItems) insert(item Item) {
 type PriorityQueue struct {
 	waiters     waiters
 	items       priorityItems
+	itemMap     map[Item]bool
```

Comment:
Thanks for the tip, after catching up with the empty struct and its underlying width (0 bytes), it definitely makes sense to use it in this context.

The purpose of the set is indeed to keep track of existing elements, since duplicates are not allowed. I can picture some cases where supporting duplicates can be handy though, but this is out of the scope of this change. Off the top of my head, would it make sense to have a flag on `PriorityQueue` to determine whether or not this is permitted?